py/desispec/workflow/os.py

```diff
@@ -0,0 +1,117 @@
+
+
+import os
+import numpy as np
+import glob
+import json
+import psutil
+
+
+pathjoin = os.path.join
+
+def listpath(*args):
+    # return np.sort(os.listdir(pathjoin(*args))).tolist()
+    if os.path.exists(pathjoin(*args)):
+        srtlist = sorted(os.listdir(pathjoin(*args)))
+        if '.DS_Store' in srtlist:
+            srtlist.remove('.DS_Store')
+        return srtlist
+    else:
+        return []
+
+
+def globpath(*args):
+    # return np.sort(glob.glob(pathjoin(*args))).tolist()
+    if os.path.exists(pathjoin(*args)):
+        srtlist = sorted(glob.glob(pathjoin(*args)))
+        if '.DS_Store' in srtlist:
+            srtlist.remove('.DS_Store')
+        return srtlist
+    else:
+        return []
+
+
+def get_file_list(filename, doaction=True):
+    if doaction and filename is not None and os.path.exists(filename):
+        output = np.atleast_1d(np.loadtxt(filename, dtype=int)).tolist()
+    else:
+        output = []
+    return output
+
+
+########################
+### Helper Functions ###
+########################
+
+
+
+def get_skipped_expids(expid_filename, skip_expids=True):
+    return get_file_list(filename=expid_filename, doaction=skip_expids)
+
+def find_newexp(night, fileglob, known_exposures):
+    """
+    Check the path given for new exposures
+    """
+    datafiles = sorted(glob.glob(fileglob))
+    newexp = list()
+    for filepath in datafiles:
+        expid = int(os.path.basename(os.path.dirname(filepath)))
+        if (night, expid) not in known_exposures:
+            newexp.append((night, expid))
+
+    return set(newexp)
+
+def get_json_dict(reqpath):
+    req_dict = {}
+    if os.path.isfile(reqpath):
+        with open(reqpath, 'r') as req:
+            req_dict = json.load(req)
+    return req_dict
+
+def give_relevant_details(verbose_output, non_verbose_output=None, verbosely=False):
```

Comment:
Just guessing from the name, but it smells like this is better handled by log.DEBUG vs. log.INFO and setting the verbosity by the logger loglevel.

Reply:
This was created out of the mindset of "repeated code should be put into a function" and this was repeated numerous times in the workflow. Since then the uses has decreased, and maybe it is no longer worth using. But for now I'm going to leave it.

It is different because it gives different output depending on if the verbose flag is set or not. The two different outputs could be sent to each of those respectively, but it doesn't eliminate the need for this function.